filters/auth/oidc.go

```diff
@@ -763,6 +763,12 @@ func (f *tokenOidcFilter) tokenClaims(ctx filters.FilterContext, oauth2Token *oa
 		return nil, "", requestErrorf("claims do not contain sub")
 	}
 
+	exp, ok := tokenMap["exp"].(float64)
+	if ok {
+		f.validity = time.Until(time.Unix(int64(exp), 0))
```

Comment:
I think we should check here also if the duration is valid, say if > time.Minute

Reply:
I agree otherwise we would DDoS the IAM endpoint if we have a 0 for example  due to some bug

Reply:
Good point! Adjusted :slightly_smiling_face: